logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -100,8 +101,9 @@ public Queue(Settings settings) {
             }
             this.dirPath = queueDir.toRealPath();
         } catch (final IOException ex) {
-            throw new IllegalStateException(ex);
+            throw new QueueRuntimeException(QueueExceptionMessages.CREATING_QUEUE_DIR_ERROR, ex);
```

Comment:
this should still be an `IllegalStateException`. It is not a runtime error and indicates that the queue could not be created due to an unexpected illegal state.